galsim/wfirst/wfirst_wcs.py

```diff
@@ -39,31 +39,40 @@
-                      108.450, 108.450])
-sca_yc_mm = np.array([0., 199.250, 149.824, 242.630, 188.250, 138.824, 231.630, 164.923, 115.498,
-                      208.304, 199.250, 149.824, 242.630, 188.250, 138.824, 231.630, 164.923,
-                      115.498, 208.304])
-# Nominal center of FPA in this coordinate frame, in mm and as an angle.
+# projected onto the sky. Note that the origin is centered on the FPA, but to get coordinates at the
+# center of the payload axes on the telescope boresight, we can simply add fpa_xc_mm and fpa_yc_mm.
+# Since the SCAs are 1-indexed, these arrays have a non-used entry with index 0.  i.e., the maximum
+# SCA is 18, and its value is in sca_xc_mm[18].  Units are mm.  The ordering of SCAs was swapped
+# between cycle 5 and 6, with 1<->2, 4<->5, etc. swapping their y positions.  Gaps between the SCAs
+# were also modified, and the parity was flipped about the f2 axis to match the FPA diagrams.
+sca_xc_mm = np.array([0., 21.44, 21.44, 21.44, 67.32, 67.32, 67.32, 113.20, 113.20,
+                      113.20, -21.44, -21.44, -21.44,  -67.32, -67.32, -67.32, -113.20,
+                      -113.20, -113.20])
+sca_yc_mm = np.array([0., -12.88,  30.00,  72.88, -24.88,  18.00,  60.88,  -42.88,     0.0,
+                      42.88,  -12.88, 30.00, 72.88, -24.88, 18.00, 60.88, -42.88,
+                      0.00,    42.88])
+# Nominal center of FPA from the payload axis in this coordinate system, in mm and as an angle.
 fpa_xc_mm = 0.0
-fpa_yc_mm = 199.250
+# This calculation is using the WIM page on the WFIRST_MCS_WFC_Zernike_and_Field_Data_160610.xlsm
+# spreadsheet on the https://wfirst.gsfc.nasa.gov/science/Inst_Ref_Info_Cycle6.html page.
+fpa_yc_mm = focal_length*(np.tan(0.46107*galsim.degrees/galsim.radians) - \
+                                   np.tan(-0.038903*galsim.degrees/galsim.radians))
 xc_fpa = np.arctan(fpa_xc_mm/focal_length)*galsim.radians
 yc_fpa = np.arctan(fpa_yc_mm/focal_length)*galsim.radians
+# Now move the sca coordinates to be with respect to the payload axis location.
+sca_xc_mm += fpa_xc_mm
+sca_yc_mm += fpa_yc_mm
 
 # The next array contains rotation offsets of individual SCA Y axis relative to FPA f2 axis. Same
 # sign convention as theta_fpa. These represent mechanical installation deviations from perfect
 # alignment and are ideally zero. These will be measured during focal plane integration and
 # testing.
 sca_rot = np.zeros_like(sca_xc_mm)
 
-# Rotation of focal plane axes relative to payload axes: positive CCW rotation of the f2 axis
-# relative to -Y_pl, and of f1 relative to +X_pl.
-theta_fpa = 32.5*galsim.degrees
+# Rotation of focal plane axes relative to payload axes: this is expressed as a CCW rotation of the
+# f2 axis relative to -Y_pl, and of f1 relative to +X_pl.  In cycle 5, this was around +30 degrees,
+# but in cycle 6 there is a 90 degrees CCW rotation with respect to that.
+theta_fpa = -60.0*galsim.degrees
```

Comment:
Is it exactly -60 now?  Your comment implies that it should be 32.5 - 90 = -57.5.  But maybe the update also included 32.5 -> 30.

Reply:
Yes, it's exactly -60 (confirmed with Chris and others).  Comment should have a twiddle; it's around 90, not exactly.  Will push this change in next round of commits.